runtime-manager/src/managers/execution_engine_manager.rs

```diff
@@ -172,33 +177,52 @@ fn parse_incoming_buffer(
     let mut incoming_buffer_hash = INCOMING_BUFFER_HASH.lock()?;
 
     // First, make sure there is an entry in the hash for the TLS session.
-    if incoming_buffer_hash.get(&tls_session_id).is_none() {
-        incoming_buffer_hash.insert(tls_session_id, Vec::new());
+    // If not, this means we are receiving the first chunk of the protocol
+    // buffer.
+    let mut input = if incoming_buffer_hash.get(&tls_session_id).is_none() {
+        // Extract the protocol buffer's total length as u64
+        let input_data = input.split_off(8);
+        let mut expected_length: [u8; 8] = [0; 8];
+        expected_length.copy_from_slice(input.as_slice());
```

Comment:
Should this be `input.as_slice()` or `input_data.as_slice()`?  Should `input_data` not be called `expected_length_bytes` or similar?

Reply:
After the call to `split_off()`, `input` is reduced to its first 8 bytes and `input_data` contains the rest of the vector.
Agreed that the variable names don't make that very clear. Also I'm doing something ugly with `input` getting assigned the return of an `if` block. I'll make some  changes